tests/errors/pure_warnings.py

```diff
@@ -0,0 +1,63 @@
+# mode: error
+# tag: warnings
+
+import cython
+import typing
+from cython.cimports.libc import stdint
+
+
+def main():
+    foo1: typing.Tuple = None
+    foo1: typing.Bar = None
+    foo2: Bar = 1  # warning
+    foo3: int = 1
+    foo4: cython.int = 1
+    foo5: stdint.bar = 5  # warning
+    foo6: object = 1
+    foo7: cython.bar = 1  # error
+    foo8: (1 + x).b
+    foo9: mod.a.b
+    foo10: func().b
+    with cython.annotation_typing(False):
+        foo8: Bar = 1
+        foo9: stdint.bar = 5
+        foo10: cython.bar = 1
+
+
+@cython.cfunc
+def bar() -> cython.bar:
```

Comment:
Question: I have found out that unrecognised type in return value is yielding an error. So now with this PR we are raising an error + new warning for the same line. Not sure if it is best behaviour.

Reply:
This only applies to C functions, where defining a type probably means that the user really wants that exact type there. An error seems right in that case. Let's keep both for now and remove the warning if it really starts annoying us.